cfg/aws/aws_sdk_logging.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package aws
+
+import (
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/influxdata/wlog"
+)
+
+var sdkLogLevel aws.LogLevelType = aws.LogOff
+
+// SetSDKLogLevel sets the global log level which will be used in all AWS SDK calls.
+// Example usage: set `debug = true` and `aws_sdk_log_level = LogDebug` in config.json.
+func SetSDKLogLevel(wLogLevel wlog.Level, sdkLogLevelString string) {
+	if wLogLevel != wlog.DEBUG {
+		sdkLogLevel = aws.LogOff
+		return
+	}
+	var temp aws.LogLevelType = aws.LogOff
+	// Hard coded strings that match actual variable names in the AWS SDK.
+	// I don't expect these names to change, or their meaning to change.
+	// This code may need updates if the SDK adds more levels (unlikely).
+	// Please note that the levels are a bit field that is OR'd together.
+	if strings.Contains(sdkLogLevelString, "LogDebug") {
+		temp |= aws.LogDebug
+	}
+	if strings.Contains(sdkLogLevelString, "LogDebugWithSigning") {
+		temp |= aws.LogDebugWithSigning
+	}
+	if strings.Contains(sdkLogLevelString, "LogDebugWithHTTPBody") {
+		temp |= aws.LogDebugWithHTTPBody
+	}
+	if strings.Contains(sdkLogLevelString, "LogDebugWithRequestErrors") {
+		temp |= aws.LogDebugWithRequestErrors
+	}
+	if strings.Contains(sdkLogLevelString, "LogDebugWithEventStreamBody") {
+		temp |= aws.LogDebugWithEventStreamBody
+	}
+
+	sdkLogLevel = temp
+}
+
+// SDKLogLevel returns the single global value so it can be used in all
+// AWS SDK calls scattered throughout the Agent.
+func SDKLogLevel() *aws.LogLevelType {
+	return &sdkLogLevel
+}
+
+// SDKLogger implements the aws.Logger interface.
+type SDKLogger struct {
+}
+
+// Log is the only method in the aws.Logger interface.
+func (SDKLogger) Log(args ...interface{}) {
+	// Always use debug logging level.
```

Comment:
Are you sure?  Doesn't the SDK ever log errors?

Reply:
Good question.
I am sure.
Here is the doc:
https://docs.aws.amazon.com/sdk-for-go/api/aws/#LogLevelType
The only levels that currently exist all begin with `LogDebug`.
There is no option to just `LogError` from the AWS SDK.

I do see this:
```
    // LogDebugWithRequestErrors states the SDK should log when service requests fail
    // to build, send, validate, or unmarshal.
    LogDebugWithRequestErrors
```
But that is still `LogDebug...`.